index.mjs

```diff
@@ -126,9 +150,15 @@ const CustomElementMixin = (superclass) => class extends superclass {
       if (slot) {
         if (slot.name) {
           if (!namedSlots[slot.name]) namedSlots[slot.name] = { slotNode: slot, contentToSlot: [] }
+          if (child['dataset']) {
+            child.dataset.slotted = true
```

Comment:
Use of `dataset.slotted` to track what elements are slotted in.
Needs some additions for text content. 
These additions will also need to be added to Enhance SSR during slot expansion so this will work with SSR'd elements as well.